src/pseudo/PspUpf.jl

```diff
@@ -0,0 +1,220 @@
+using LinearAlgebra
+using Interpolations: linear_interpolation
+using PseudoPotentialIO: load_upf
+
+struct PspUpf{T,IT} <: NormConservingPsp
+    Zion::Int                             # Ionic charge (Z - valence electrons)
+    lmax::Int                             # Maximal angular momentum in the non-local part
+    
+    rgrid::Vector{T}                      # Radial grid
+    drgrid::Vector{T}                     # Radial grid derivative / integration factor
+    
+    vloc::Vector{T}                       # Local potential on the radial grid
+    vloc_interp::IT                       # Local potential interpolator
+    
+    projs::Vector{Vector{Vector{T}}}      # Kleinman-Bylander β projectors: projs[l+1][i]
+    projs_interp::Vector{Vector{IT}}      # Projector interpolator
+    h::Vector{Matrix{T}}                  # Projector coupling coefficients per AM channel: h[l+1][i1,i2]
+    
+    pswfcs::Vector{Vector{T}}             # Pseudo-atomic wavefunctions
+    pswfc_ang_moms::Vector{Int}           # Angular momenta of the pseudo-atomic wavefunctions
+    pswfc_occs::Vector{T}                 # Occupations of the pseudo-atomic wavefunctions
+
+    rhoatom::Vector{T}                    # Pseudo-atomic charge density
+
+    rvlocpzdr::Vector{T}                  # (r_i V_{loc}(r_i) + Z) dr_i
+    r2projsdr::Vector{Vector{Vector{T}}}  # r_j^2 β_{li}(r_j) dr_i
+    
+    identifier::String                    # String identifying the PSP
+end
+
+"""
+    PspUpf(Zion::Number, lmax::Number, rgrid::Vector, drgrid::Vector, vloc::Vector,
+           vloc_interp, projs::Vector{Vector{Vector}}, projs_interp, h::Vector{Matrix},
+           pswfcs::Vector{Vector}, pswfc_ang_moms::Vector, pswfc_occs::Vector,
+           rhoatom::Vector, rvlocpzdr::Vector, r2projsdr::Vector{Vector{Vector}};
+           identifier="")
```

Comment:
Maybe we should move to https://github.com/JuliaDocs/DocStringExtensions.jl to avoid this sort of stuff.

Reply:
Seems like a great idea!

Reply:
oh yes, separate pr